cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -225,27 +230,40 @@ def _add_login_nodes_pool_auto_scaling_group(self):
         auto_scaling_group = autoscaling.CfnAutoScalingGroup(
             self,
             f"{self._login_nodes_stack_id}-AutoScalingGroup",
+            auto_scaling_group_name=f"{self._login_nodes_stack_id}-AutoScalingGroup",
             launch_template=launch_template_specification,
             min_size=str(self._pool.count),
             max_size=str(self._pool.count),
             desired_capacity=str(self._pool.count),
             target_group_arns=[self._login_nodes_pool_target_group.node.default_child.ref],
             vpc_zone_identifier=self._pool.networking.subnet_ids,
         )
-
-        self._add_lifecycle_hook(auto_scaling_group)
+        self.terminating_lifecycle_hook = self._add_terminating_lifecycle_hook(auto_scaling_group)
+        self.launching_lifecycle_hook = self._add_launching_lifecycle_hook(auto_scaling_group)
 
         return auto_scaling_group
 
-    def _add_lifecycle_hook(self, auto_scaling_group):
+    def _add_terminating_lifecycle_hook(self, auto_scaling_group):
         return autoscaling.CfnLifecycleHook(
             self,
-            "LoginNodesASGLifecycleHook",
+            "LoginNodesASGLifecycleHookTerminating",
             auto_scaling_group_name=auto_scaling_group.ref,
             lifecycle_transition="autoscaling:EC2_INSTANCE_TERMINATING",
+            lifecycle_hook_name=f"{self._login_nodes_stack_id}-LoginNodesTerminatingLifecycleHook",
             heartbeat_timeout=self._pool.gracetime_period * 60,
         )
 
+    def _add_launching_lifecycle_hook(self, auto_scaling_group):
+        return autoscaling.CfnLifecycleHook(
+            self,
+            "LoginNodesASGLifecycleHookLaunching",
+            auto_scaling_group_name=auto_scaling_group.ref,
+            lifecycle_hook_name=f"{self._login_nodes_stack_id}-LoginNodesLaunchingLifecycleHook",
+            lifecycle_transition="autoscaling:EC2_INSTANCE_LAUNCHING",
+            default_result="ABANDON",
+            heartbeat_timeout=600,
```

Comment:
Nitpick: do we want this hardcoded or controllable through a config parameter (even non customer facing) ?

This could help for tests...

Reply:
If you're referring to the fact that we are hardwiring the lifecycle hook name: it requires paradoxically some refactoring because there is a circular dependency between ASG and launch template. Solving the circular dependency would require a refactoring that it's not worth it for the beta.

If you are referring to the customizability of the timeout I like your proposal, but it should be at most a dev settings and i would introduce this customizability in another PR not considering it beta blocking.

What do you think?

Reply:
:+1:

I was referring to the timeout ..but I agree we can do it in another PR